Add unittest to components

worfsmile opened the pull request.

worfsmile committed:
new file:   __init__.py
	modified:   tests/pygame_gui/test_components.py

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

worfsmile committed:
new file:   tests/__init__.py
	new file:   tests/pygame_gui/__init__.py

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,38 @@
+import sys
+import os
+
+print(os.path.dirname(__file__))
```
Don't print useless infomation in test

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,38 @@
+import unittest
```
Sort imports.

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,38 @@
+print(os.path.dirname(__file__))
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
+
+from src.pygame_gui.components.root import *
```
Import concrete class. For examle
`from src.pygame_gui.components.root import Root`
You should never use `import *`

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,38 @@
+        root = Root()
+        self.assertIsInstance(root, Root)
+
+class TestBase(unittest.TestCase):
```
Root depends on Base, you should test Base first.

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,38 @@
+import os
+
+print(os.path.dirname(__file__))
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
```
Don't change system path.

worfsmile committed:
modified:   src/pygame_gui/components/base.py
	modified:   src/pygame_gui/utils.py
	modified:   tests/pygame_gui/test_components.py

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,26 @@
+from src.pygame_gui.components.base import Base
+from src.pygame_gui.components.root import Root
+        
+class TestRoot(unittest.TestCase):
```
Use one test class to test one original class

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

EclidFundgue commented on tests/pygame_gui/test_components.py:
```diff
@@ -0,0 +1,36 @@
+
+
+class TestBase(unittest.TestCase):
+    def setUp(self):
```
Why do you init pygame when testing `Base`

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

worfsmile committed:
modified:   tests/pygame_gui/test_components.py

EclidFundgue commented on __init__.py:
WTF is this

worfsmile committed:
deleted:    __init__.py
